src/main/java/com/iota/iri/Snapshot.java

```diff
@@ -26,6 +31,7 @@
 
     public static final Map<Hash, Long> initialState = new HashMap<>();
     public static final Snapshot initialSnapshot;
+    public final ReadWriteLock rwlock = new ReentrantReadWriteLock();
```

Comment:
see https://stackoverflow.com/questions/26087738/performance-of-reentrantreadwritelock-read-lock and the follow up https://stackoverflow.com/questions/26070041/synchronizing-searches-and-modifications/26075089#26075089 for why we should rather use `StampedLock` here.